new_dawn_server/users/api/resources.py

```diff
@@ -344,10 +344,31 @@ def _get_taken_info(self, bundle):
             bundle.data["taken_requested_from_me"] = False
 
 
+    def _get_taken_by_info(self, bundle):
```

Comment:
This code will be called when we retrieve the user's profile. We want this field to always present in user profile. With that said, we should just check if from this user there's an already taken sent to someone else (since ALREADY_TAKEN is bi-directional). We should only have one field called "taken_by" instead of two, which contains who I'm taken with. You can just use user_from__id__exact without specifying user_to_id__exact.